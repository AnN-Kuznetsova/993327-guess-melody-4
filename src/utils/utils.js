const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const noop = () => {
  // do nothing
};


export {
  extend,
  noop,
};
