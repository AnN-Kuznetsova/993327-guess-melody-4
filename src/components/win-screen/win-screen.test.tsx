import * as React from "react";
import * as renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import {WinScreen} from "./win-screen";
import {history} from "../../history";
import {noop} from "../../utils/utils";


const props = {
  questionsCount: null,
  mistakesCount: null,
  onReplayButtonClick: noop,
};


describe(`Render WinScreen`, () => {
  it(`Should match with snapshot`, () => {
    props.questionsCount = 5;
    props.mistakesCount = 0;
    let winScreenSnapshot = renderer.create(
        <Router history={history} >
          <WinScreen {...props} />
        </Router>
    ).toJSON();
    expect(winScreenSnapshot).toMatchSnapshot();

    props.questionsCount = 5;
    props.mistakesCount = 1;
    winScreenSnapshot = renderer.create(
        <Router history={history} >
          <WinScreen {...props} />
        </Router>
    ).toJSON();
    expect(winScreenSnapshot).toMatchSnapshot();

    props.questionsCount = 3;
    props.mistakesCount = 2;
    winScreenSnapshot = renderer.create(
        <Router history={history} >
          <WinScreen {...props} />
        </Router>
    ).toJSON();
    expect(winScreenSnapshot).toMatchSnapshot();
  });
});
