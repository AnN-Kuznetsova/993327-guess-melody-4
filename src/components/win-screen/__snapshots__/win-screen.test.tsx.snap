// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render WinScreen Should match with snapshot 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    5
     вопросов и совершили 
    0
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`Render WinScreen Should match with snapshot 2`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    4
     вопросов и совершили 
    1
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`Render WinScreen Should match with snapshot 3`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    1
     вопросов и совершили 
    2
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;
